### ISSUE TITLE
Too slow cached queries

### ISSUE THREAD

**Issue by koutoulastha** [VahidN/EFCoreSecondLevelCacheInterceptor#35]
Hello,
I'm testing EFCoreSecondLevelCacheInterceptor with MySql.Data.EntityFrameworkCore 8.0.20 connector and I have noticed that cached queries take 150-200 ms to execute while I believe it shouldn't take more than 1-2 ms. I have used both the built-in and the EasyCaching.InMemory memory cache providers.
Is this normal behavior?

Thank you in advance

**Comment by RyanThomas73** [comment]
I see similar performance for large data sets. If I understand correctly this library just caches the db response but the `IQueryable` expression tree compilation and object hydration still happen every time:

*Not cached yet:*
EF Processes IQueryable Expression Tree -> EF Translates Query to SQl and builds Relational Command -> Relational Command is executed and response is read -> EF processes the response and hydrates new instances of the objects

*Cached:*
EF Processes IQueryable Expression Tree -> EF Translates Query to sql (if it was unable to cache the sql before) -> Interceptor returns cached db response -> EF processes the response and hydrates new instances of the objects

**Comment by koutoulastha** [comment]
Thank you for your reply,

I tested with large and small data sets, I also tried detaching the entities using AsNoTracking and I got the same behavior. Cache invalidation is a great feature but it really hurts the performance of my application and possibly our database.

EntityFrameworkCore.Cacheable is the package I'm trying to replace. Although it works very well, I can't get it to work on EntityFrameworkCore 3.1 and MySql.Data.EntityFrameworkCore 8.0.20. 
As a last resort I can always implement my own in memory cache and add the entities in it but I was hoping for a better, cleaner solution.
